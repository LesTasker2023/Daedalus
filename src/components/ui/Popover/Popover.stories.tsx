import type { Meta, StoryObj } from '@storybook/nextjs';
import { Popover } from './Popover';

const meta: Meta<typeof Popover> = {
  title: 'UI/Popover',
  component: Popover,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    variant: {
      control: { type: 'select' },
      options: ['default', 'destructive', 'outline', 'secondary', 'ghost', 'link'],
    },
    size: {
      control: { type: 'select' },
      options: ['default', 'sm', 'lg', 'icon'],
    },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Popover',
  },
};

export const Variants: Story = {
  render: () => (
    <div className="flex gap-4">
      <Popover variant="default">Default</Popover>
      <Popover variant="destructive">Destructive</Popover>
      <Popover variant="outline">Outline</Popover>
      <Popover variant="secondary">Secondary</Popover>
      <Popover variant="ghost">Ghost</Popover>
      <Popover variant="link">Link</Popover>
    </div>
  ),
};

export const Sizes: Story = {
  render: () => (
    <div className="flex gap-4 items-center">
      <Popover size="sm">Small</Popover>
      <Popover size="default">Default</Popover>
      <Popover size="lg">Large</Popover>
      <Popover size="icon">‚≠ê</Popover>
    </div>
  ),
};
