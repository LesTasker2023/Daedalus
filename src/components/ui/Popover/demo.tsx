'use client';

import { Popover, PopoverContent, PopoverTrigger } from '.';
import { Button } from '../Button';

export default function PopoverDemo() {
  return (
    <div className="space-y-6 p-6 max-w-2xl mx-auto">
      <div className="space-y-2">
        <h1 className="text-2xl font-bold">Popover Component</h1>
        <p className="text-muted-foreground">Click-triggered content overlay</p>
      </div>

      <div className="space-y-4">
        <div className="space-y-2">
          <h3 className="text-lg font-semibold">Basic Usage</h3>
          <div className="border rounded-lg p-4 bg-background">
            <Popover>
              <PopoverTrigger asChild>
                <Button variant="outline">Open Popover</Button>
              </PopoverTrigger>
              <PopoverContent>
                <div className="p-2">
                  <h4 className="font-semibold">Popover Content</h4>
                  <p className="text-sm text-muted-foreground">
                    This content appears in a popover.
                  </p>
                </div>
              </PopoverContent>
            </Popover>
          </div>
        </div>

        <div className="space-y-2">
          <h3 className="text-lg font-semibold">Interactive</h3>
          <div className="border rounded-lg p-4 bg-background">
            <Popover>
              <PopoverTrigger asChild>
                <Button>Settings</Button>
              </PopoverTrigger>
              <PopoverContent>
                <div className="p-4">
                  <h4 className="font-semibold">Settings Panel</h4>
                  <p className="text-sm text-muted-foreground">Configure your preferences here.</p>
                </div>
              </PopoverContent>
            </Popover>
          </div>
        </div>
      </div>
    </div>
  );
}
