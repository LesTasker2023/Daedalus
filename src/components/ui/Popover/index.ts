export { default as PopoverDemo } from './demo';
export type { PopoverProps } from './Popover';
export * from './PopoverReal';
