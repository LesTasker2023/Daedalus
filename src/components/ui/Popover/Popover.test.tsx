import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Popover } from './Popover';

describe('Popover', () => {
  it('renders without crashing', () => {
    render(<Popover>Test</Popover>);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('applies variant classes correctly', () => {
    render(<Popover variant="destructive">Test</Popover>);
    expect(screen.getByRole('button')).toHaveClass('bg-destructive');
  });

  it('applies size classes correctly', () => {
    render(<Popover size="lg">Test</Popover>);
    expect(screen.getByRole('button')).toHaveClass('h-11');
  });

  it('handles custom className', () => {
    render(<Popover className="custom-class">Test</Popover>);
    expect(screen.getByRole('button')).toHaveClass('custom-class');
  });

  it('forwards ref correctly', () => {
    const ref = { current: null };
    render(<Popover ref={ref}>Test</Popover>);
    expect(ref.current).toBeInstanceOf(HTMLButtonElement);
  });

  // Accessibility tests
  it('has proper accessibility attributes', () => {
    render(<Popover aria-label="Test button">Test</Popover>);
    expect(screen.getByRole('button')).toHaveAttribute('aria-label', 'Test button');
  });

  it('handles keyboard navigation', () => {
    render(<Popover>Test</Popover>);
    const button = screen.getByRole('button');
    button.focus();
    expect(button).toHaveFocus();
  });
});
