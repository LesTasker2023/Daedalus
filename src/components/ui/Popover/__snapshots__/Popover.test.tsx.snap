// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popover > Snapshots > matches default snapshot 1`] = `
<button
  class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
  data-testid="popover"
>
  Test content
</button>
`;

exports[`Popover > Snapshots > matches disabled state snapshot 1`] = `
<button
  class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
  data-testid="popover"
  disabled=""
>
  Test content
</button>
`;

exports[`Popover > Snapshots > matches hover state snapshot 1`] = `
<button
  class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
  data-testid="popover"
>
  Test content
</button>
`;
